### PR TITLE
Feat/user balance

### PR DESCRIPTION
**Pull request by sorrelbri**
Balance page styling is in need of a lot of work.

I'll need to spend some time looking at `makeStyles` to improve.

### CODE REVIEW

File: client/src/pages/Balance.js [hunk 1]
```diff
@@ -0,0 +1,100 @@
...
+  Grid,
+  Typography,
+} from "@material-ui/core";
+
+const BalancePage = ({ state, dispatch }) => {
+  const user = useContext(userContext);
+  const [topUp, setTopUp] = useState(2);
+
+  const handleClick = ({ target, ...e }) => {
+    // e.prototype.preventDefault();
```
Comment (skassam21) [inline client/src/pages/Balance.js:21]: maybe remove this?

File: client/src/pages/Balance.js [hunk 2]
```diff
@@ -0,0 +1,100 @@
...
+              style={{ display: "flex" }}
+              wrap="nowrap"
+            >
+              <Grid item>
+                <Button
+                  onClick={handleClick}
+                  id="decr"
+                  variant="contained"
+                  size="small"
+                  disabled={state.loading}
```
Comment (skassam21) [inline client/src/pages/Balance.js:65]: disabled when topUp === 0 too?

File: client/src/pages/Balance.js [hunk 3]
```diff
@@ -0,0 +1,100 @@
...
+
+  return (
+    <Background solid>
+      <Container component="main" maxWidth="xs">
+        <CssBaseline />
+        <StyledPaper mt={20}>
+          {/* Page Heading */}
+          <PageHeader>Your Balance:</PageHeader>
+          <Typography
+            component="p"
```
Comment (skassam21) [inline client/src/pages/Balance.js:37]: I rarely change components in Typography - any reason why you opted to do that?

File: client/src/pages/Balance.js [hunk 4]
```diff
@@ -0,0 +1,100 @@
...
+                  {topUp}
+                </Typography>
+              </Grid>
+              <Grid item>
+                <Button
+                  onClick={handleClick}
+                  id="incr"
+                  disabled={state.loading}
+                >
+                  +
```
Comment (skassam21) [inline client/src/pages/Balance.js:81]: Consider using icon buttons - https://material-ui.com/components/material-icons/

File: client/src/pages/Balance.js [hunk 5]
```diff
@@ -0,0 +1,100 @@
...
+  Typography,
+} from "@material-ui/core";
+
+const BalancePage = ({ state, dispatch }) => {
+  const user = useContext(userContext);
+  const [topUp, setTopUp] = useState(2);
+
+  const handleClick = ({ target, ...e }) => {
+    // e.prototype.preventDefault();
+    const { id } = target.closest("button");
```
Comment (skassam21) [inline client/src/pages/Balance.js:22]: any reason not to do e.target.id?

File: client/src/pages/Balance.js (hunk 2 above)
Comment (sorrelbri) [reply client/src/pages/Balance.js:65]: totally

File: client/src/pages/Balance.js (hunk 5 above)
Comment (sorrelbri) [reply client/src/pages/Balance.js:22]: The `Button` component creates a button element that wraps some spans, which means the event target is actually the span that acts as the button label. 
```html
<button class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-sizeSmall" 
  tabindex="0" type="button" id="decr"
>
  <span class="MuiButton-label">-</span>
  <span class="MuiTouchRipple-root"></span>
</button>
```

Comment (sorrelbri) [reply client/src/pages/Balance.js:22]: It looks like the `.MuiTouchRipple-root` has some click and focus handlers that spawn additional child nodes

File: client/src/pages/Balance.js (hunk 3 above)
Comment (sorrelbri) [reply client/src/pages/Balance.js:37]: I'm not familiar with MaterialUI, but it seems really easy to fall out of line with a11y best practice. Seemed safer to err on the side of being overly imperative.

**Pull request merged**